### PR TITLE
Implemented Integrity NoBibtexFieldChecker

### PR DESCRIPTION
**Pull request by grimes2**
Background: Converting to Biblatex format changes journal to journaltitle. Because journaltitle is a Biblatex field only, Integrity check now reports a problem for a journaltitle field in a BibTeX mode database.

- [x] Change in CHANGELOG.md described
- [x] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [x] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/integrity/NoBibtexFieldChecker.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+
+public class NoBibtexFieldChecker implements Checker {
+
+    private final BibDatabaseContext bibDatabaseContextEdition;
+
+
+    public NoBibtexFieldChecker(BibDatabaseContext bibDatabaseContext) {
+        this.bibDatabaseContextEdition = Objects.requireNonNull(bibDatabaseContext);
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/integrity/NoBibtexFieldChecker.java]: To make copy and paste easier: Just rename the variable to `bibDatabseContext`.

File: src/main/java/net/sf/jabref/logic/integrity/NoBibtexFieldChecker.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public List<IntegrityMessage> check(BibEntry entry) {
+        Optional<String> value = entry.getField(FieldName.JOURNALTITLE);
+        if (!value.isPresent()) {
+            return Collections.emptyList();
+        }
+
+        //BibTeX
+        if (!bibDatabaseContextEdition.isBiblatexMode() && value.isPresent()) {
+            return Collections.singletonList(
+                    new IntegrityMessage(Localization.lang("no BibTeX field"), entry, FieldName.JOURNALTITLE));
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/integrity/NoBibtexFieldChecker.java]: Maybe it is better to reword it and say: BibLaTeX field only?

File: src/main/java/net/sf/jabref/logic/integrity/NoBibtexFieldChecker.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    private final BibDatabaseContext bibDatabaseContextEdition;
+
+
+    public NoBibtexFieldChecker(BibDatabaseContext bibDatabaseContext) {
+        this.bibDatabaseContextEdition = Objects.requireNonNull(bibDatabaseContext);
+    }
+
+    /**
+     * BibLaTeX package documentation (Section 4.9.1):
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/integrity/NoBibtexFieldChecker.java]: Adjust javadoc

**Comment by grimes2** [comment]
Done.

**Comment by Siedlerchr** [comment]
Good! I merge it in!

**Pull request merged**